packages/strapi-design-system/src/Select/Select.js

```diff
@@ -0,0 +1,196 @@
+import React, { Children, cloneElement, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import DropdownIcon from '@strapi/icons/FilterDropdownIcon';
+import CloseAlertIcon from '@strapi/icons/CloseAlertIcon';
+import { SelectButton } from './SelectButton';
+import { Field, FieldHint, FieldLabel, FieldError } from '../Field';
+import { Popover } from '../Popover';
+import { Stack } from '../Stack';
+import { Text } from '../Text';
+import { Row } from '../Row';
+import { genId } from '../helpers/genId';
+import { SelectList } from './SelectList';
+import { SelectButtonWrapper, IconBox, CaretBox } from './components';
+
+export const Select = ({
+  label,
+  id,
+  children,
+  placeholder,
+  onChange,
+  value,
+  hint,
+  error,
+  disabled,
+  clearLabel,
+  ...props
+}) => {
+  const idRef = useRef(id || genId());
+  const listRef = useRef(null);
+  const buttonRef = useRef(null);
+  const containerRef = useRef(null);
+  const [expanded, setExpanded] = useState(false);
+
+  const labelId = `label-${idRef.current}`;
+  const contentId = `content-${idRef.current}`;
+
+  const handleTrigger = (direction = 'down') => {
+    setExpanded((s) => !s);
+
+    /**
+     * Make sure the focus is sent when React has finished its rendering phase
+     */
+    setTimeout(() => {
+      if (!listRef.current) return;
+
+      const lastSelected = listRef.current.querySelector('[aria-selected="true"]');
+      const options = listRef.current.querySelectorAll('[role="option"]');
+
+      if (direction === 'up') {
+        const lastOption = lastSelected || options[options.length - 1];
+
+        if (lastOption) {
+          lastOption.focus();
+        }
+      } else {
+        const firstOption = lastSelected || options[0];
+
+        if (firstOption) {
+          firstOption.focus();
+        }
+      }
+    }, 0);
+  };
+
+  const handleEscape = () => {
+    setExpanded(false);
+
+    buttonRef.current.focus();
+  };
+
+  const handleClear = () => {
+    if (disabled) return;
+
+    onChange(undefined);
```

Comment:
I'm not sure why it would be necessary? What is the use case (just in case we can find a better solution)

Reply:
Well when you want to clear your data you have several ways:
- Creating an option with the undefined option (it can be needed in some cases)
- Using the clear button

In both case I think it is useful for such complex component to have some metas indicating programmatically what the user did. React Select as such behaviours and I think it's very useful.

Reply:
I'll add a "onClear" prop :)